tests/test-system-management.py

```
@@ -454,6 +455,241 @@ def test_real_regional_reads_use_an_isolated_private_bus(self):
+        with self.catalog(), mock.patch.object(provider, "close_locale_process",
+                side_effect=provider.SnapshotFailure("timeout", "Fixture cleanup failure")):
+            with self.assertRaises(provider.SnapshotFailure) as caught:
+                provider.read_locale_choices()
+        self.assertEqual(caught.exception.code, "timeout")
+
+    def test_cleanup_failure_cannot_hide_pending_process_termination(self):
+        for number in (signal.SIGTERM, signal.SIGINT, signal.SIGHUP):
+            for during_cleanup in (False, True):
+                handler = signal.getsignal(number)
+                def cleanup(_process):
+                    if during_cleanup:
```

Ruff (0.16.3)

[warning] Function definition does not bind loop variable `during_cleanup`

(B023)